### PR TITLE
#3076436 - Add configuration option to send user details

### PR DESCRIPTION
**Pull request by safalliajoseph**
d.o: https://www.drupal.org/project/elastic_apm/issues/3076436

### CODE REVIEW

**Comment by safalliajoseph** [comment]
@krystalcode I have added configuration option for sending user details. Could you please review.

File: src/Form/PhpAgentSettingsForm.php [hunk 1]
```diff
@@ -135,6 +135,16 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
 
+    $form['privacy'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Privacy settings'),
+    ];
+    $form['privacy']['send_user'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Send user details'),
+      '#default_value' => $config->get('phpAgent.privacy.send_user'),
+    ];
```
Comment (krystalcode) [inline src/Form/PhpAgentSettingsForm.php]: **CHANGE REQUEST**

"Separate section" meant to be a separate settings page/tab, not just separate section in the agent config.
The reasoning is that this is not really a setting of the agent, it configures which details to log like the other tab we have to configure which pages to log for. There is no setting in the PHP agent for that.

**Review by krystalcode — CHANGES_REQUESTED** [review]

File: src/Form/PhpAgentSettingsForm.php (hunk 1 above)
Comment (safalliajoseph) [reply src/Form/PhpAgentSettingsForm.php]: @krystalcode I have added privacy to a separate tab. Could you please review

**Comment by krystalcode** [comment]
Merged on D.org

**Pull request closed without merging**